eventstore/query.go

```diff
@@ -69,8 +72,11 @@ func (q *Query) OrderByDesc(field string) *Query {
 // pointers with the correct model type. If the slice isn't empty, will be emptied.
 func (t *Txn) Find(res interface{}, q *Query) error {
 	valRes := reflect.ValueOf(res)
-	if valRes.Kind() != reflect.Ptr || valRes.Elem().Kind() != reflect.Slice {
-		panic("result should be a slice")
+	if valRes.Kind() != reflect.Ptr || // Should be a pointer
```

Comment:
The `res` parameter can't be anything. It should be a concrete type for the query considering the model being queried. Now it checks that is what's expected... since if that check doesn't hold, the code will panic in line: `		resSlice = reflect.Append(resSlice, unsorted[i])`

TBH, these checks has some cost... nothing is free in particular with refect. If the client is sending the wrong thing, maybe it should panic because that's a serious error here. Shouldn't be expected by any means... so maybe these checks can be deleted and make the client responsible to use it correctly.